Set up data links on dataset page

alexgleith opened the pull request:
* Adds a list of data URIs to download TIFs

Jeremy, I'm sure I'm missing something. Let me know what I can do to make this more robust?

jeremyh commented on cubedash/summary/_stores.py:
```diff
@@ -1561,6 +1563,21 @@ def get_dataset_footprint_region(self, dataset_id):
+        Get a representation of the data URIs so that they
+        can be linked to to download them.
+        """
+        dataset_doc = serialise.from_doc(dataset.metadata_doc, skip_validation=True)
```
This parser only reads eo3 docs, which is why the tests of older datasets are failing.

Simple option may be to check if it's eo3 with [eodatasets3.utils.is_doc_eo3(doc)](https://github.com/GeoscienceAustralia/eo-datasets/blob/8f5088ce0281b9b4b465fcbfb71fdd58d3c70a83/eodatasets3/utils.py#L144-L145) and not show the list otherwise.

jeremyh replied:
If we want to support everything, a more-thorough option could be to extract the basic [eo-to-eo3 logic](https://github.com/opendatacube/datacube-explorer/blob/ee63dcb381e623dba87c07375d5b592eea952ec2/cubedash/_stac.py#L638-L640) into a function, and run that first. (but ideally we might move it into eodatasets proper)

jeremyh commented on cubedash/summary/_stores.py:
```diff
@@ -1561,6 +1563,21 @@ def get_dataset_footprint_region(self, dataset_id):
+
+        uri_list = {
+            name: urljoin(dataset_location, m.path)
+            for name, m in dataset_doc.measurements.items()
```
We probably want to include `dataset_doc.accessories.items()` too. 

They are all the dataset files that aren't measurements: non-odc metadata, thumbnails, checksums, etc.

alexgleith replied:
Done

jeremyh reviewed (commented):
Nice idea!

alexgleith committed:
Add data URIs if a product uses EO3-compatible metadata

alexgleith committed:
Fix formatting

alexgleith committed:
Merge branch 'develop' into add-data-uris-to-dataset-page

alexgleith committed:
Replace math/path

jeremyh commented on cubedash/summary/_stores.py:
```diff
@@ -1561,6 +1564,34 @@ def get_dataset_footprint_region(self, dataset_id):
+
+        if is_doc_eo3(dataset.metadata_doc):
+            dataset_doc = serialise.from_doc(dataset.metadata_doc, skip_validation=True)
+            dataset_location = (
+                dataset_doc.locations[0] if dataset_doc.locations else None
+            )
```
When running everything's coming back as a relative path (which are broken as they're relative to Explorer's webpage url). Presumably it's because this field is null.

(The raw indexed `dataset.metadata_doc` column doesn't contain locations)

We'll need to grab locations from the original Dataset object instead:
```suggestion
            dataset_location = (
                dataset.uris[0] if dataset.uris else None
            )
```

jeremyh commented on cubedash/templates/dataset.html:
```diff
@@ -4,6 +4,8 @@
 
 {% set extent_geojson = dataset_footprint.__geo_interface__ %}
 
+{% set data_uris = data_uris %}
```
Does that do something in jinja?

alexgleith replied:
Oh, probably not. I was just copying the patterns from before it!

jeremyh reviewed (commented):
The paths are quirky for me when running locally -- see below.

(I'd also like to fiddle with the css/styles of the panel at some point if I get the time, to make it more consistent with others... but it wouldn't block this pr.)

alexgleith committed:
Update S3 URIs to be HTTPS URIs

alexgleith committed:
Merge branch 'add-data-uris-to-dataset-page' of github.com:opendatacube/datacube-explorer into add-data-uris-to-dataset-page

jeremyh committed:
Dataset paths markup: combine with locations

Put them on the same panel, and show dataset paths
for each location rather than just the first.

jeremyh committed:
Consistent config file name

jeremyh committed:
Fix vertical rhythm, files header

jeremyh committed:
Call them 'links' not 'files'

I guess.

jeremyh committed:
Adjust link header visual weight
